### PR TITLE
Backend support for homepage hero logo image override

### PR DESCRIPTION
**Pull request by DamienBurkeWN**
### Setup
1. Checkout this branch. It's based on the summary PR to keep config changes down.
2. `./drush.wrapper @docker config:import`
3. Clear the drupal cache

### Testing
1. In the CMS, view the homepage. Note lovely new preview layout.
2. Edit the homepage and set a hero logo image override and save/publish. View the page and note lovely preview now includes lovely overridden logo.
3. In the /graphql form use the following query:
```
{
  route(path: "/") {
    node {
      title
      heroLogoOverride {
        url
        altText
      }
    }
  }
}
```
...and verify the URL and altText are correct.
4. Edit the homepage, remove the logo override and save/publish. Rerun the graphql query and verify that heroLogoOverride now returns null.

### CODE REVIEW

File: backend/src/config/sync/core.entity_form_display.node.homepage.default.yml [hunk 1]
```diff
@@ -112,16 +116,24 @@ content:
     region: content
     settings: {  }
     third_party_settings: {  }
+  field_hero_logo_override:
+    type: image_image
+    weight: 34
+    region: content
+    settings:
+      progress_indicator: throbber
```
Comment (DamienBurkeWN) [inline backend/src/config/sync/core.entity_form_display.node.homepage.default.yml:124]: You may snigger here

File: backend/src/modules/nc_system/src/Entity/Node/Homepage.php [hunk 2]
```diff
@@ -68,6 +69,24 @@ protected function getHeroImagesResolved(): array {
...
 
+  protected function getHeroLogoOverride(): ?array {
+    try {
+      $image_value = $this->get('field_hero_logo_override')->getValue();
+      if (!empty($image_value)) {
+        $image = $image_value[0];
+        $file = File::load($image['target_id']);
+        return empty($file) ? null : [
+          'id' => $image['target_id'],
+          'url' => $file->createFileUrl(false),
```
Comment (DamienBurkeWN) [inline backend/src/modules/nc_system/src/Entity/Node/Homepage.php:80]: Today's lesson is don't bother googling for solutions to Drupal stuff. Just read the source code.

File: backend/src/config/sync/core.entity_view_display.node.service_landing_page.default.yml [hunk 3]
```diff
@@ -125,11 +125,11 @@ content:
     region: content
   field_show_summary:
     type: boolean
-    label: above
```
Comment (DamienBurkeWN) [inline backend/src/config/sync/core.entity_view_display.node.service_landing_page.default.yml:128]: Ah, oops - this was the stuff from the suggestion I made yesterday

File: backend/src/config/sync/core.entity_view_display.paragraph.promotional_banner.default.yml [hunk 4]
```diff
@@ -18,39 +20,39 @@ mode: default
 content:
   field_body:
     type: text_default
-    label: above
```
Comment (DamienBurkeWN) [inline backend/src/config/sync/core.entity_view_display.paragraph.promotional_banner.default.yml:21]: Changes in here to improve preview of promo banner when viewing home page in CMS

File: backend/src/config/sync/core.entity_view_display.paragraph.service_links.default.yml [hunk 5]
```diff
@@ -13,15 +13,15 @@ mode: default
 content:
   field_service_landing_page:
     type: entity_reference_label
-    label: above
```
Comment (DamienBurkeWN) [inline backend/src/config/sync/core.entity_view_display.paragraph.service_links.default.yml:16]: Changes in here to improve preview of services list when viewing home page in CMS

File: backend/src/modules/nc_system/src/GraphQLFieldResolver.php [hunk 6]
```diff
@@ -115,6 +115,9 @@ public static function resolveMediaImage($image, $width = NULL, $height = NULL)
     $mid = $image['target_id'];
     $media = Media::load($mid);
 
+    if (!$media) {
```
Comment (DamienBurkeWN) [inline backend/src/modules/nc_system/src/GraphQLFieldResolver.php:118]: Introduced to save a future dev some pain, maybe

File: backend/src/themes/nc_seven/css/ckeditor/page--homepage.css [hunk 7]
```diff
@@ -0,0 +1,118 @@
+/* Homepage content type tweaks, currently all applying view mode only */
```
Comment (DamienBurkeWN) [inline backend/src/themes/nc_seven/css/ckeditor/page--homepage.css:1]: All this to make the homepage preview look nicer

File: backend/src/modules/nc_system/src/Entity/Node/Homepage.php (hunk 2 above)
Comment (chrisrhymes) [reply backend/src/modules/nc_system/src/Entity/Node/Homepage.php:80]: Sounds awesome... Glad you got it sorted!

File: backend/src/modules/nc_system/src/GraphQLFieldResolver.php (hunk 6 above)
Comment (chrisrhymes) [reply backend/src/modules/nc_system/src/GraphQLFieldResolver.php:118]: Will it appear in the drupal logs?

Comment (DamienBurkeWN) [reply backend/src/modules/nc_system/src/GraphQLFieldResolver.php:117]: It will now :)

**Review by chrisrhymes — APPROVED** [review]

**Pull request merged**